apps/gitness/src/pages-v2/repo/repo-list.tsx

```diff
@@ -17,28 +16,27 @@ export default function ReposListPage() {
   const { setRepositories, page, setPage } = useRepoStore()
 
   /* Query and Pagination */
-  const [query] = useDebouncedQueryState('query')
+  const [query, setQuery] = useQueryState('query')
   const [queryPage, setQueryPage] = useQueryState('page', parseAsInteger.withDefault(1))
 
   const {
-    data: { body: repoData, headers } = {},
+    data: { body: repoData } = {},
     refetch,
-    isLoading,
+    isFetching,
     isError,
     error
   } = useListReposQuery(
     {
-      queryParams: { page, query },
+      queryParams: { page, query: query ?? '' },
       space_ref: `${space}/+`
     },
-    { retry: false }
-  )
-
-  useEffect(() => {
-    if (repoData) {
-      setRepositories(repoData, headers)
+    {
+      retry: false,
+      onSuccess: data => {
```

Comment:
do we need this change? since `onSuccess` is deprecated, let's see if it still works the same with a useEffect